TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -652,6 +668,7 @@ public override bool Visit(DataDefinition dataDefinition)
 
             var commonDataDataDefinitionCodeElement = dataDefinitionEntry as CommonDataDescriptionAndDataRedefines;
 
+            var codeElementType = dataDefinitionEntry.Type;
```

Comment:
Bonus: can be moved in inner scope.
Same thing for line 676 `var levelNumberValue` can be moved inside next `else` block

Reply:
Done